rcl_lifecycle/include/rcl_lifecycle/transition_map.h

```diff
@@ -51,6 +114,25 @@ rcl_lifecycle_register_state(
   rcl_lifecycle_state_t state,
   const rcl_allocator_t * allocator);
 
+/// Regiter a transition
+/*
+ * This function registers a new transition in the transition map.
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | Yes
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | Yes
+ *
+ * \param[in] transition_map
```

Comment:
Add brief description saying along the lines of "transition_map to be modified". Here and below